cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -1031,6 +1031,37 @@
 
 
 
+;; TODO: LLVM lowers this as not+clz for i32/i64. We should do the same.
```

Comment:
I don't understand this TODO comment. As far as I understand we can't use `not` on the input without checking if it was negative. But isn't that exactly what the rule for `lower_cls` already does?

Reply:
Oops, I think I messed up slightly here. I looked at rust's output for `u32::leading_ones`, but that has slightly different semantics. LLVM's output for the equivalent operation is still better than our lowering, I'll update the comment with the godbolt link.

```
example::cls_u64:
        bgez    a0, .LBB4_2
        not     a0, a0
.LBB4_2:
        clz     a0, a0
        ret
```

Unfortunately it requires a branch, which sort of prevents us from easly implementing this in ISLE :disappointed: .